### DISCUSSION TITLE
How would I get information about why a streaming connection closed?

### DISCUSSION THREAD

**Discussion by isiah-lloyd** [node-fetch/node-fetch#1458] [Q&A]
I am streaming live audio using `node-fetch`. I ran into an issue where my connection was getting closed after a certain amount of time on a certain VPS provider. I am thinking perhaps the IP was on a block list, but I really had no idea how to diagnose it. How would I get more information about why a connection closed?

**Example Code**

```js
const audioStream = await fetch(station.streamDownloadURL, {
  headers: new Headers({'Icy-Metadata': '1'})
})
const metaInt = audioStream.headers.get('icy-metaint')
let resource: AudioResource

if (metaInt) {
  const spliceMetadata = new SpliceMetadata(parseInt(metaInt), this.updateCurrentPlaying.bind(this))
  audioStream.body.pipe(spliceMetadata)
  resource = createAudioResource(spliceMetadata)
} else {
  resource = createAudioResource(station.streamDownloadURL)
}

this.PLAYER.play(resource)
```

This code pipes the body through a transformer and then transcodes the audio with FFmpeg.

### Your Environment

| software         | version
| ---------------- | -------
| node-fetch       |  2.6.2
| node             | 16.10
| npm              | 8.1.4
| Operating System | Ubuntu 18.04

**Comment by jimmywarting** [comment]
Hmm, hard to tell... maybe this could work?

```js
audioStream.body.pipe(spliceMetadata).on('error', console.error)
```

fyi, body is going to be a whatwg stream later in v4... just a heads up, still need node stream, then you will be able to do: `stream.readable.from(res.body)`

another way to deal with it could be to use async await...
then you can handle it the same way weather or not it's a node or whatwg stream.

```js
try {
  for await (const uint8 of audioStream.body) {
    spliceMetadata.write(uint8)
  } 
} catch (err) {
  console.log(err)
}
```
